nengo/config.py

```diff
@@ -84,6 +84,10 @@ def __repr__(self):
         return "<%s[%s]{%s}>" % (type(self).__name__,
                                  self._configures.__name__, ", ".join(params))
 
+    def __setstate__(self, state):
+        for k, v in state.items():
+            setattr(self, k, v)
```

Comment:
Without it a bunch of tests fail:
```
________________________________________________ TestCopyNetwork.test_pickle_roundtrip _________________________________________________
[gw3] darwin -- Python 2.7.12 /Volumes/Home/blubb/.virtualenvs/nengodev/Python.app/Contents/MacOS/Python
self = <nengo.tests.test_copy.TestCopyNetwork object at 0x1112b6650>

    def test_pickle_roundtrip(self):
        original = self.create_original()
>       cp = pickle.loads(pickle.dumps(original))

nengo/tests/test_copy.py:121:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
nengo/config.py:50: in __getattr__
    return self.get_param(key).get_default(self)
nengo/config.py:100: in get_param
    if key in self._extraparams:
nengo/config.py:50: in __getattr__
    return self.get_param(key).get_default(self)
nengo/config.py:100: in get_param
    if key in self._extraparams:
nengo/config.py:50: in __getattr__
    return self.get_param(key).get_default(self)
!!! Recursion detected (same locals & position)
__________________________________________ TestCopyNetwork.test_unpickling_warning_in_network __________________________________________
[gw3] darwin -- Python 2.7.12 /Volumes/Home/blubb/.virtualenvs/nengodev/Python.app/Contents/MacOS/Python
self = <nengo.tests.test_copy.TestCopyNetwork object at 0x1112b61d0>

    def test_unpickling_warning_in_network(self):
        original = self.create_original()
        pkl = pickle.dumps(original)
        with nengo.Network():
            with warns(NotAddedToNetworkWarning):
>               pickle.loads(pkl)

nengo/tests/test_copy.py:129:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <nengo.utils.testing.warns object at 0x112588110>
type = <type 'exceptions.RuntimeError'>
value = RuntimeError('maximum recursion depth exceeded',)
traceback = <traceback object at 0x1127533f8>

    def __exit__(self, type, value, traceback):
        if not any(r.category is self.warning_type for r in self.record):
>           self._pytest.fail("DID NOT WARN")
E           Failed: DID NOT WARN

nengo/utils/testing.py:205: Failed
_________________________________________ TestCopyEnsemble.test_unpickling_warning_in_network __________________________________________
[gw1] darwin -- Python 2.7.12 /Volumes/Home/blubb/.virtualenvs/nengodev/Python.app/Contents/MacOS/Python
self = <nengo.tests.test_copy.TestCopyEnsemble object at 0x10b73e450>

    def test_unpickling_warning_in_network(self):
        original = self.create_original()
        pkl = pickle.dumps(original)
        with nengo.Network():
            with warns(NotAddedToNetworkWarning):
>               pickle.loads(pkl)

nengo/tests/test_copy.py:129:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <nengo.utils.testing.warns object at 0x10e9b6090>, type = None
value = None, traceback = None

    def __exit__(self, type, value, traceback):
        if not any(r.category is self.warning_type for r in self.record):
>           self._pytest.fail("DID NOT WARN")
E           Failed: DID NOT WARN

nengo/utils/testing.py:205: Failed
========================================== 3 failed, 762 passed, 45 skipped in 41.83 seconds ==========================================
```

But I have to dig dipper to tell you the reason.